### ISSUE TITLE
Use from a controller?

### ISSUE THREAD

**Issue by imclint21** [serilog-contrib/SerilogSinksInMemory#36]
Hi,

I try to use from a controller but `InMemorySink.Instance.LogEvents` seems always empty, does I need to configure any DI?

Regards

**Comment by sandermvanvliet** [comment]
Would you be able to share a code sample of how you’re attempting to do this?

**Comment by imclint21** [comment]
Hi, yes sure, this is my service:

```csharp
public class LogService(InMemorySink memorySink)
{
	public IEnumerable<Log> GetAll() => memorySink.LogEvents
		.DistinctBy(logEvent => logEvent.MessageTemplate.Text)
		.Select(logEvent => new Log
		{
			Date = logEvent.Timestamp,
			Level = logEvent.Level switch
			{
				LogEventLevel.Information => SeverityLevels.Information,
				LogEventLevel.Warning => SeverityLevels.Warning,
				LogEventLevel.Error or LogEventLevel.Fatal => SeverityLevels.Error,
				_ => SeverityLevels.Trace
			},
			Message = logEvent.RenderMessage()
		});
}
```

**Comment by imclint21** [comment]
Do you need more stuffs @sandermvanvliet?

**Comment by sandermvanvliet** [comment]
Without seeing a repro case it's a bit difficult for me to determine where this may be breaking down.

I see you're passing in a `InMemorySink` instance but I don't know how you've configured Serilog and are consuming that Serilog `ILogger` instance elsewhere.

One thing that _could_ happen is that the logger is disposed before you're calling `GetAll()` from your `LogService`. I've seen before that all log entries are gone then.

**Comment by imclint21** [comment]
I will try to give you more part of the code, and before check what you said

**Comment by sandermvanvliet** [comment]
Mixing and matching file and code configuration is a recipe for disaster so I’m not entirely surprised this happened 😅

**Comment by imclint21** [comment]
Yeah, didn't really knew how this lib was working, and didn't really pay attention :)
